src/sink.c

```diff
@@ -3,16 +3,26 @@
 
 /* Known sink constructors */
 extern sink* init_stream_sink(const sink_config_stream*, const statsite_config*);
+extern sink* init_http_sink(const sink_config_http*, const statsite_config*);
 
 int init_sinks(sink** sinks, statsite_config* config) {
     for(sink_config* sc = config->sink_configs; sc != NULL; sc = sc->next) {
-        if (sc->type == SINK_TYPE_STREAM) {
+        switch(sc->type) {
+        case SINK_TYPE_STREAM:
+        {
             sink* actual_sink = init_stream_sink((sink_config_stream*)sc, config);
-            if (*sinks) {
-                actual_sink->next = *sinks;
-            }
+            actual_sink->next = *sinks;
             *sinks = actual_sink;
-        } else {
+            break;
+        }
+        case SINK_TYPE_HTTP:
+        {
+            sink* actual_sink = init_http_sink((sink_config_http*)sc, config);
+            actual_sink->next = *sinks;
```

Comment:
Yeah, I'll look at some macros for single LL intrusive operations. Its mostly straightforward. I'll open an issue for this one as a number of places could benefit.